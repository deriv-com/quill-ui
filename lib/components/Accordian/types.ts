import { ReactNode } from "react";
import { AccordionVariants } from ".";
import { TRegularSizes } from "@types";

export type AccordionProps = {
    id?: string;
    className?: string;
    title?: string;
    subtitle?: string;
    icon?: ReactNode;
    color?: string;
    textSize?: TRegularSizes;
    expanded?: boolean;
    divider?: AccordionDivider;
    disabled?: boolean;
    contentClassname?: string;
    expandedColor?: TExpandedColor;
    content?: () => ReactNode;
    customContent?: () => ReactNode;
    isFlush?: boolean;
    onExpand?: (e: boolean, i: string) => void;
};
type TExpandedColor = "white" | "gray";
export interface AccordionTabProps extends Omit<AccordionProps, "content"> {
    tab?: Tab;
    className?: string;
    variant?: keyof AccordionVariants;
    content: {
        className?: string;
        data: AccordionProps[][];
    };
    multiCollapse?: boolean;
}

export type Tab = {
    align?: TabAlign;
    data: TabData[];
};

export type TabData = {
    id: number;
    title: string;
};

export type TabAlign = "center" | "end";

export type AccordionDivider = "none" | "both" | "bottom";
