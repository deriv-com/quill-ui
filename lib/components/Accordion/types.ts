import { ReactNode } from "react";
import { AccordionVariants } from ".";
import { TRegularSizes } from "@types";

export type AccordionProps = {
    id?: string;
    className?: string;
    title?: string;
    subtitle?: string;
    icon?: ReactNode;
    color?: string;
    textSize?: TRegularSizes;
    expanded?: boolean;
    divider?: AccordionDivider;
    disabled?: boolean;
    contentClassname?: string;
    content?: () => ReactNode;
    customContent?: () => ReactNode;
    onExpand?: (e: boolean, i: string) => void;
};

export interface AccordionTabProps extends Omit<AccordionProps, "content"> {
    className?: string;
    variant?: keyof AccordionVariants;
    content: {
        className?: string;
        data: AccordionProps[][];
    };
    multiCollapse?: boolean;
}

export type AccordionDivider = "none" | "both" | "bottom";
