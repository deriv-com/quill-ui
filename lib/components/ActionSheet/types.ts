import { ComponentPropsWithoutRef } from "react";
import { ExcludeAllNull, TLeftOrRight } from "../../types";

interface OpenType {
    isOpen?: boolean | undefined;
    onOpen?: () => void;
}

interface actionSheetRootCVA {
    position?: TLeftOrRight;
    show?: boolean;
}

interface actionSheetFooterCVA {
    alignment?: "vertical" | "horizontal";
}

export interface RootProps
    extends ComponentPropsWithoutRef<"div">,
        ExcludeAllNull<actionSheetRootCVA>,
        OpenType {
    onClose?: () => void;
    type?: "modal" | "non-modal";
    expandable?: boolean;
}

export type RootPosition = RootProps["position"];

interface ActionType {
    content: string;
    onAction: () => void;
}

export interface FooterProps
    extends ComponentPropsWithoutRef<"div">,
        actionSheetFooterCVA {
    primaryAction?: ActionType;
    secondaryAction?: ActionType;
    shouldCloseOnPrimaryButtonClick?: boolean;
    shouldCloseOnSecondaryButtonClick?: boolean;
}

export type FooterAlignment = FooterProps["alignment"];
