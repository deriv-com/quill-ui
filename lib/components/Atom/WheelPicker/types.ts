import { THorizontalPosition } from "@types";
import { HTMLAttributes, ReactNode } from "react";

export type TWheelTypeSelectItem = {
    value: string | number;
    label?: ReactNode;
};
export interface WheelPickerProps extends HTMLAttributes<HTMLElement> {
    data: TWheelTypeSelectItem[];
    selectedValue: string | number;
    setSelectedValue: (value: string | number) => void;
    handleKeyDown?: (e: React.KeyboardEvent) => void;
    isFocused?: boolean;
    dropDownItemClassName?: string;
    containerClassName?: string;
    listClassName?: string;
    position?: THorizontalPosition | undefined;
    containerHeight?: string;
    disabled?: boolean;
}
export interface WheelPickerContainerProps
    extends Omit<
        WheelPickerProps,
        "data" | "setSelectedValue" | "selectedValue"
    > {
    data: TWheelTypeSelectItem[][];
    inputValues?: (string | number)[];
    setInputValues?: (index: number, value: string | number) => void;
    children?: React.ReactNode;
    close?: () => void;
    setSelectedValue?: (value: string | number) => void;
    containerHeight?: string;
}
export interface TimeWheelPickerContainerProps
    extends Omit<
        WheelPickerProps,
        "data" | "setSelectedValue" | "selectedValue"
    > {
    is12Hour: boolean;
    children?: React.ReactNode;
    minutesInterval?: number;
    hoursInterval?: number;
    locale?: string;
    close?: () => void;
    setSelectedValue?: (value: string | number) => void;
    startTimeIn24Format?: string;
    endTimeIn24Format?: string;
    selectedTime?: string;
    setSelectedTime?: (value: string) => void;
    data?: TWheelTypeSelectItem[][];
    containerHeight?: string;
}
export type TTypeOfWheel = "Time" | "Generic" | "Date";
