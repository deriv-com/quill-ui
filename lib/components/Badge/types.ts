import { TRegularSizes } from "@types";

export type TBadgeColorVariants = "success" | "warning" | "danger";
export type TBadgeVariant = "status" | "notification";

export type BadgeProps = {
    variant?: TBadgeVariant;
    size?: TRegularSizes;
    color?: TBadgeColorVariants;
    className?: string;
    label?: string;
    badgeClassName?: string;
};
