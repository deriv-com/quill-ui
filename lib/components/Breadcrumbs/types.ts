import { TMediumSizes } from "@types";
import { ReactNode } from "react";

export interface LinkProps {
    content: ReactNode;
    href: string;
    target?: "_blank" | "_self" | "_parent" | "_top";
}

export type BreadcrumbProps = {
    size?: TMediumSizes;
    links?: LinkProps[];
    className?: string;
};
