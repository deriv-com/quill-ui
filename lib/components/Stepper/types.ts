import { TypographyProps } from "@components/Typography/base";

type TEnum = { [key: string]: number };
export type TTypographySizes = Extract<TypographyProps['size'], 'sm' | 'md' | 'lg'>;

export type TBasicStepperProps = {
    currentStep: number | TEnum;
    Icon?: ({ size }: { size: TTypographySizes }) => React.ReactNode;
    size?: TTypographySizes;
    labels: string[];
    lineSize?: string;
    orientation?: string;
};
