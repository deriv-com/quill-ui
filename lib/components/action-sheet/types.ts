import { ComponentPropsWithoutRef } from "react";

type OpenType =
    | {
          isOpen?: boolean;
          onOpen: () => void;
      }
    | {
          isOpen?: undefined;
          onOpen?: () => void;
      };

export type RootProps = ComponentPropsWithoutRef<"div"> &
    OpenType & {
        onClose?: () => void;
        type?: "modal" | "non-modal";
        expandable?: boolean;
    };

type ActionType = {
    content: string;
    onAction: () => void;
};

export type FooterProps = ComponentPropsWithoutRef<"div"> & {
    primaryAction?: ActionType;
    secondaryAction?: ActionType;
};

export type FooterAlignment = FooterProps;
