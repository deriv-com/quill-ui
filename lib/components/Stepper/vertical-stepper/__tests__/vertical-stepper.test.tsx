import { render, screen } from "@testing-library/react";
import { BasicStepper } from "../../index";
import { Steps } from "../../types";

describe("BasicStepper component", () => {

    const props = {
        currentStep: 1,
        labels: ['Step 1', 'Step 2', 'Step 3'],
        allSteps: Steps,
    };

    it("renders steps correctly", () => {
        render(<BasicStepper {...props} />);
        expect(screen.getByText("Step 1")).toBeInTheDocument();
        expect(screen.getByText("Step 2")).toBeInTheDocument();
    });

    it("marks active step correctly", () => {
        render(<BasicStepper {...props} />);
        const activeCircle = screen.getAllByTestId("dt-step-circle")[0];
        expect(activeCircle).toHaveClass("step-circle--active");
    });

    it("disables future steps correctly", () => {
        render(<BasicStepper {...props} currentStep={0} size={"sm"} />);
        const futureStep = screen.getAllByTestId("dt-step-line")[0];
        screen.debug()
        expect(futureStep).toHaveClass("step-line--disabled");
    });

    it("renders custom icons correctly", () => {
        const CustomIcon = () => <div data-testid="custom-icon">Custom</div>;
        render(<BasicStepper {...props} Icon={CustomIcon} size={"sm"} />);
        expect(screen.getByTestId("custom-icon")).toBeInTheDocument();
    });

    it("matches snapshot", () => {
        const { container } = render(<BasicStepper {...props} size={"sm"} />);
        expect(container).toMatchSnapshot();
    });

});
