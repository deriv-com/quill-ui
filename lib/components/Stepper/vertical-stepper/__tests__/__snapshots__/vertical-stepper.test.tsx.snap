// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicStepper component matches snapshot 1`] = `
<div>
  <div
    class="steps-container"
  >
    <div
      class="step"
    >
      <div
        class="step-circle__pointer"
      >
        <div
          class="step-circle step-circle__size-sm step-circle--active"
          data-testid="dt-step-circle"
        >
          <svg
            fill="var(--semantic-color-slate-solid-surface-frame-low)"
            height="24"
            role="img"
            viewBox="0 0 14 24"
            width="11"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M13.688 7.313a.964.964 0 0 1 0 1.406l-8 8a.964.964 0 0 1-1.407 0l-4-4a.964.964 0 0 1 0-1.406.964.964 0 0 1 1.406 0L5 14.593l7.281-7.28a.964.964 0 0 1 1.406 0"
              />
            </g>
            <defs>
              <clippath
                id="24498f3e3cdcc3012813186d8185f82c__a"
              >
                <path
                  d="M0 0h14v24H0z"
                />
              </clippath>
            </defs>
          </svg>
        </div>
        <div
          class="step-line step-line--size-md"
          data-testid="dt-step-line"
        />
      </div>
      <span
        class="quill-typography quill-typography__body-text__size--sm__weight--bold__decoration--default quill-typography__color--default step-circle__label-sm"
      >
        Step 1
      </span>
    </div>
    <div
      class="step"
    >
      <div
        class="step-circle__pointer"
      >
        <div
          class="step-circle step-circle__size-sm step-circle--inactive"
          data-testid="dt-step-circle"
        />
      </div>
      <span
        class="quill-typography quill-typography__body-text__size--sm__weight--bold__decoration--default quill-typography__color--default step-circle__label-sm"
      >
        Step 2
      </span>
    </div>
  </div>
</div>
`;
